datafusion/src/optimizer/simplify_expressions.rs

```diff
@@ -453,7 +450,7 @@ mod tests {
     fn test_simplify_and_and_false() -> Result<()> {
         let expr =
             binary_expr(lit(ScalarValue::Boolean(None)), Operator::And, lit(false));
-        let expr_eq = lit(false);
+        let expr_eq = null;
```

Comment:
I think the original answer of `false` is correct. Here is the result in postgres:

```
alamb=# select null and false;
 ?column? 
----------
 f
(1 row)

alamb=# 
```